pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -107,6 +107,7 @@ stdenv.mkDerivation ({
     ${optionalString (hasActiveLibrary && hyperlinkSource) "export PATH=${hscolour}/bin:$PATH"}
 
     configureFlags="--verbose --prefix=$out --libdir=\$prefix/lib/\$compiler --libsubdir=\$pkgid $configureFlags"
+    configureFlags+=" --with-gcc=$CC"
```

Comment:
https://gist.github.com/gridaphobe/46c1164af4788c9fe0d6

The issue seems to be that cabal only looks for a `gcc` binary, which we don't have in the clang stdenv.